### DISCUSSION TITLE
December 12th 2023 Release

### DISCUSSION THREAD

**Discussion by rphlo** [karttapullautin/karttapullautin#33] [Announcements]
This release includes the rust binary along the perl script that are both necessary to run a conversion job.

The functions ported to rust are:
  - blocks
  - makecliffs
  - makevegenew
  - pointdxfcrop
  - polylinedxfcrop
  - smoothjoin ***NEW IN THIS RELEASE***
  - xyz2contours
  - xyzknolls
  - and the handling of the multithreading

The rust functions are generating the same output as their perl equivalent but faster.

This also handle a vegetation only mode with the `vegeonly=1` flag.

<hr /><em>This discussion was created from the release <a href='https://github.com/rphlo/rusty-pullauta/releases/tag/20231212a'>12th December 2023 Release</a>.</em>

**Comment by rphlo** [comment]
The goal for 2024 will be to slowly continue to replace the remaining perl functions.

**Reply by stefankinell** [reply]
Thanks for all the good work this year! Truly appreciated!

**Reply by jaluzdi** [reply]
I totally agree with Stefan !!!
Thanks for all the good coding of this project !!!
The new rusty-pullauta is amazingly fast compared to the old one, without detracting from the great work of Jarkko.
And as Manu Jurado (one of the bigs brains behind @mapant_es) told you, it has already been tested processing part of the second coverage of spanish lidar data. And it will help us more in the future with greater density points per square meter in the third coverage of spanish lidar.

**Reply by rphlo** [reply]
I am also greatful for your projects Mapant.es  and Gokartor.se.
I do not have myself a direct use of kartta pullatin but knowing it is used in your projects is what motivates me to continue improving it.

**Comment by rphlo** [comment]
On my test file, My mac mini M1 is now generating a map in 21 seconds with the Rusty version, while the fully Perl version was taking 1 minute 25 seconds.
